src/test/ui/span/issue-42595.stderr

```diff
@@ -0,0 +1,16 @@
+error[E0368]: binary assignment operation `+=` cannot be applied to type `&mut {integer}`
+  --> $DIR/issue-42595.rs:15:5
+   |
+15 |     map.entry("e").or_insert(0) += 1;
+   |     ---------------------------^^^^^
+   |     |
+   |     cannot use `+=` on type `&mut {integer}`
```

Comment:
I'm a little confused as to the intent of this message. Is it trying to just point at the `+=` (which seems correct)? If so, the underline-like arrow beneath the rest of the line looks really odd to me. Might just not be used to it though.. nor do I have any ideas as to what else we could do.